routing/blinding.go

```diff
@@ -124,15 +133,38 @@ func NewBlindedPaymentPathSet(paths []*BlindedPayment) (*BlindedPaymentPathSet,
 	// replacement, so our target pub key in this case just remains the
 	// real introduction node ID.
 	for _, path := range paths {
-		if len(path.BlindedPath.BlindedHops) != 1 {
-			continue
-		}
+		pathLength := len(path.BlindedPath.BlindedHops)
```

Comment:
since this loop does more now, the comment above it only applies to the first `if pathLength == 1` part - so consider moving the comment inside to where it is relevant. Or expand the comment above to explain the dummy hop thing